ts/features/itwallet/common/components/ItwCredentialCard.tsx

```diff
@@ -0,0 +1,207 @@
+import {
+  Badge,
+  Body,
+  HSpacer,
+  IOColors,
+  Tag
+} from "@pagopa/io-app-design-system";
+import React from "react";
+import { Image, ImageSourcePropType, StyleSheet, View } from "react-native";
+import I18n from "../../../../i18n";
+import { CredentialType } from "../utils/itwMocksUtils";
+
+export type ItwCredentialStatus = "valid" | "pending" | "expiring" | "expired";
+
+type PreviewProps = {
+  isPreview: true;
+  data?: never;
+};
+
+type DataProps = {
+  isPreview?: false;
+  data: ReadonlyArray<string>;
+};
+
+type BaseProps = {
+  credentialType: CredentialType;
+  isMasked?: boolean;
+  status?: ItwCredentialStatus;
+};
+
+export type ItwCredentialCard = BaseProps & (PreviewProps | DataProps);
+
+export const ItwCredentialCard = ({
+  isMasked = false,
+  status = "valid",
+  credentialType,
+  ...props
+}: ItwCredentialCard) => {
+  const isValid = status === "valid";
+  const shouldDisplayData = !(!isValid || isMasked) && !props.isPreview;
+  const labelColor: IOColors = isValid ? "bluegreyDark" : "grey-700";
+
+  const cardBackgroundSource =
+    credentialCardBackgrounds[credentialType][isValid ? 0 : 1];
+  const statusTagProps = tagPropsByStatus[status];
+
+  return (
+    <View style={props.isPreview && styles.previewContainer}>
+      <View style={styles.cardContainer}>
+        <View style={styles.card}>
+          <Image
+            style={{ height: "100%", width: "100%" }}
+            source={cardBackgroundSource}
+            accessibilityIgnoresInvertColors={true}
+          />
+        </View>
+        <View style={styles.infoContainer}>
+          <View style={styles.header}>
+            <Body
+              color={labelColor}
+              weight="SemiBold"
+              numberOfLines={2}
+              style={{ flex: 1 }}
+            >
+              {cardLabelByCredentialType[credentialType].toUpperCase()}
+            </Body>
+            {statusTagProps && (
+              <>
+                <HSpacer size={16} />
+                <Tag {...statusTagProps} />
+              </>
+            )}
+          </View>
+          {shouldDisplayData && <CredentialData {...props} />}
+        </View>
+        {!isValid && <DigitalVersionBadge />}
+        <View
+          style={[styles.border, { borderColor: borderColorByStatus[status] }]}
+        />
+      </View>
+    </View>
+  );
+};
+
+const CredentialData = ({ data }: DataProps) => (
+  <View style={styles.personalInfo}>
+    {data.map(value => (
+      <Body
+        color="bluegreyDark"
+        weight="SemiBold"
+        key={`credential_data_${value}`}
+      >
+        {value}
+      </Body>
+    ))}
+  </View>
+);
+
+const DigitalVersionBadge = () => (
+  <View style={styles.digitalVersionBadge}>
+    <Badge
+      // The space at the end is an hack to have extra padding inside the badge text
```

Comment:
🙃

Reply:
Yeah, I know, I don't like hacks either. But I didn't want to duplicate code and recreate a component that "already exists".
Do you think we should create a dedicated component to display the "digital version" badge?

Reply:
That's OK for me! The smiley was to say that I would do the same. It's okay at the moment to use what we already have.